### PR TITLE
Update non-Angular packages

### PR DESCRIPTION
**Pull request by dslucas**
The most notable change here is the upgrade to version 9 of the Firebase Web SDK.
We update the imports to the v9 compat library to keep changes small for now,
but we'll want to upgrade to the modular v9 library in the future.

See https://firebase.google.com/docs/web/modular-upgrade for more details.

Other changes are just small import or RxJS operator fixes.

### CODE REVIEW

**Comment by dslucas** [comment]
FYI I created #6 to track future work for migrating to the modular library.

File: src/server/middleware/twitter.middleware.ts [hunk 1]
```diff
@@ -83,11 +83,7 @@ export async function blockTwitterUsers(
   }
 
   const request = req.body as BlockTwitterUsersRequest;
-  const userCredential = firebase.auth.AuthCredential.fromJSON(
-    request.credential
-  ) as firebase.auth.OAuthCredential;
-
-  const response = await blockUsers(apiCredentials, userCredential, request);
+  const response = await blockUsers(apiCredentials, request);
```
Comment (rachelrosengoogle) [inline src/server/middleware/twitter.middleware.ts:86]: Just wanted to confirm that we don't need the fromJSON anymore? If I recall, there were fields that didn't translate correctly without it. See the (perhaps outdated/misplaced) comment at https://github.com/conversationai/harassment-manager/blob/9ab7aa4642040f51dda6613ab135b85693e6e17d/src/app/sheets_api.service.ts#L109 If that's still true we should move that comment here.

Comment (dslucas) [reply src/server/middleware/twitter.middleware.ts:86]: Good catch! I tested the API calls and they succeeded, so it seems like we don't need `fromJSON` anymore. I removed the comment accordingly.

Comment (rachelrosengoogle) [reply src/server/middleware/twitter.middleware.ts:86]: Great, thanks for checking!

**Review by rachelrosengoogle — APPROVED** [review]

**Pull request merged**